### DISCUSSION TITLE
Issues getting supabase auth to work with Github in Firefox

### DISCUSSION THREAD

**Discussion by alexghr** [supabase/supabase#9061] [Questions] [answered]
Hello,

I'm trying out Supabase Auth for a project where I want to sign-in with Github. The issue I'm seeing in Firefox is supabase _not_ redirecting me to my `redirectTo` and not authenticating my account _if_ I have already granted access to my Github account to Supabase. Here's the scenario:

1. I sign-in for the first time. (`supabase.auth.signIn({ provider: "github" }, { redirectTo: "http://localhost:3000/callback" })`)
2. I'm prompted by Github to grant permission to my accounts' data to Supabase
3. Supabase redirects me to my `redirectTo` URL
4. :white_check_mark: I'm signed in
5. I sign out (using `supabase.auth.signOut()`)
6. I sign-in again (`supabase.auth.signIn({ provider: "github" }, { redirectTo: "http://localhost:3000/callback" })`)
7. This time, a bunch of redirects happen
8. I land on `http://localhost:3000/` rather than `/callback` and I'm not signed in.

I have prior experience with OAuth/OIDC just not with Supabase's flavour. The redirects at step 7 are OK, with the exception that Supabase ignores `redirectTo` and takes me back to the home page.

This is a Remix app and I'm running Firefox 104.0.1 on Linux. In Chrome this flow works correctly. 

Here's what I have tried:
- :no_good_man: Calling `supabase.auth.signIn` from a button click
- :no_good_man: Calling `supabase.auth.signIn` as an effect (button click redirects to `/login` which calls `supabase.auth.signIn` as an effect)
- :no_good_man: Redirecting server-side using `supabase.auth.api.getUrlForProvider` and a service token.
- :no_good_man: fiddling with Site URL and Redirect URLs in my project settings.
- :no_good_man: fiddling with firefox icognito / containers / 3rd party cookies (they're on btw)

What I have _not_ tried:
- stripping all Remix/react and testing with plain Javascript.
- other social providers or email + password/magic link sign-in
- Firefox in other OSes

Has anyone else seen this behaviour from supabase-js?

---

PS: I'm not sure this is the right place to ask this or if it's better as an issue (but then, in which project?).

**Comment by alexghr** [comment, marked as answer]
**tl;dr** Disabling Remix's live reload feature fixes the problem.

---

I've investigated this issue a bit more and tracked it down to a problem in Remix's (the frontend framework I was using) live reload feature: https://github.com/remix-run/remix/issues/2997

It looks like their live reload script automatically reloads the page some time after it loses connection to the dev server (ie. after a redirect). This causes a race condition where if the browser does not resolve all of the Oauth redirects before this timer runs out, the page will refresh and cancel the redirect. For some reason Firefox took longer than the timeout while Chrome didn't.

This ticket can be closed now.
